cpp/src/arrow/acero/query_context.cc

```diff
@@ -51,9 +81,9 @@ size_t QueryContext::GetThreadIndex() { return thread_indexer_(); }
 size_t QueryContext::max_concurrency() const { return thread_indexer_.Capacity(); }
 
 Result<util::TempVectorStack*> QueryContext::GetTempStack(size_t thread_index) {
+  static const int64_t temp_stack_size = internal::GetTempStackSizeFromEnvVar();
   if (!tld_[thread_index].is_init) {
```

Comment:
Please move the `static` to the `if (...) {` block so that code doesn't have to always atomically call `GetTempStackSizeFromEnvVar`.

And use `ARROW_PREDIC_FALSE` on the `if` so that the compiler doesn't inline `GetTempStackSizeFromEnvVar` (cold code) keeping this function small in the binary.

Reply:
Great suggestion! Addressed.